cram_urdf_projection/cram_urdf_projection/src/low-level.lisp

```diff
@@ -339,6 +339,22 @@
         (one-gripper-action action-type (second arm) maximum-effort))
       (one-gripper-action action-type arm maximum-effort)))
 
+(defun get-gripper-length ()
+  (let* ((frame-bindings
+             (cut:lazy-car
+              (prolog:prolog
+               `(and (cram-robot-interfaces:robot ?robot)
+                     (cram-robot-interfaces:robot-tool-frame ?robot :left ?left-tool-frame)
+                     (cram-robot-interfaces:end-effector-link ?robot :left ?left-ee-frame)))))
+           (left-tcp-frame
+             (cut:var-value '?left-tool-frame frame-bindings))
+           (left-ee-frame
+             (cut:var-value '?left-ee-frame frame-bindings)))
+    (roslisp:with-fields (translation)
+        (cl-tf:lookup-transform cram-tf:*transformer* left-tcp-frame left-ee-frame)
+      (cl-tf:v-dist (cl-tf:make-3d-vector 0 0 0) translation))))
+  
```

Comment:
But actually, I'm not sure that I like this solution with TF. What if there is no TF listener. In theory, the transform between the EE and TCP should be already given in the URDF, as the TCP is fixed onto EE, the joint that connects them is always fixed (unless the person making a URDF did something weird). Please look at all the 3 URDFs of the robots that you support and find the fixed joint that connects EE to TCP of all the robots. And write here what you find out. You can use RViz TF tree to search for this. If the chain between EE and TCP is not direct (it should be EE frame -> TCP frame and not EE frame -> some other frame -> and another frame -> TCP), write down the full chain. It should be that all of them are connected directly and you can find the gripper-offset from that connection.

Reply:
So after quick research: For pr2 and hsrb the chain is: wrist_roll_link, hand_palm_link gripper_tool..
but for boxy i see *_arm_7_link -> right_arm_flange_link -> .... 11 more ... -> right_gripper_base_link -> right_gripper_tool_frame

Reply:
new solution is up